src/main/java/org/jabref/gui/LibraryTab.java

```diff
@@ -1197,4 +1205,26 @@ public String toString() {
     public LibraryTabContainer getLibraryTabContainer() {
         return tabContainer;
     }
+
+    /**
+     * Takes an entry and increments the number of times it has been viewed by 1.
+     */
+    private void incrementViewCount(BibEntry entry) {
+        MVStore mvStore = PopularityGroup.getMVStore();
+        synchronized (mvStore) {
+            MVMap<String, Integer> viewCounts = mvStore.openMap("entryViewCounts");
+            MVMap<String, Long> lastViewTimestamps = mvStore.openMap("lastViewTimestamps");
```

Comment:
Nono - the map is a class variable - no need to open the map each time for writing. 

You need to shut down the mvstore when the tab is closed.

The the lifecycle of the AiService or search for examples.